addons/draggablesprite2d/src/draggablesprite2d.gd

```diff
@@ -1,79 +1,121 @@
 @tool
-extends Area2D
+class_name DraggableSprite2D extends Area2D
+## A draggable sprite that can be dragged using the left mouse button.
 
-var sprite := Sprite2D.new()
-var collider : CollisionShape2D
-
-var is_grabbed := false :
-	set(x):
-		is_grabbed = x
-		## Emit signal if currently grabbed or not
-		if x:
-			grabbed.emit()
-		else:
-			released.emit()
-## Helps a bit to make the dragging less choppy
-var grabbed_offset := Vector2.ZERO
-## Store for the original position
-var origin := Vector2.ZERO
 
+## Emitted when the sprite is grabbed
 signal grabbed
+## Emitted when the sprite is released
 signal released
 
-## Sprite texture
-## This is a bit crap, since it can't be resized as nicely as a normal sprite2d
+
+## The input button that will be used to grab the sprite
+@export var input_method: MouseButton = MOUSE_BUTTON_LEFT
+## The texture that will be displayed by the sprite. [br]
+## Note: The texture can't be rezised as nicley as a Sprite2D.
 @export var texture : Texture2D : 
-	set(x):
-		texture = x
+	set(value):
+		texture = value
 		sprite.texture = texture
-		## Update the default collider with the shape and size of the sprite, if it exists
-		if collider and collider.shape:
-			collider.shape.size = Vector2(texture.get_width(), texture.get_height())
+		# Update the default_collider with the shape and size of the sprite, if it exists
+		update_default_collider()
 ## Whether or not the sprite should return to it's starting position when released
 @export var return_to_origin := false
 ## Whether or not it should be possible to grab the sprite
 @export var grabbable := true
 
-func _ready():
-	## Check if we need to instantiate a collider 
-	var children = get_children()
-	var needs_collider = true
-	for i in children:
-		## If a collision shape is a child, we don't need a collider
-		if i is CollisionShape2D or i is CollisionPolygon2D:
-			needs_collider = false
+
+## The Sprite node that will be used to display the texture
+var sprite := Sprite2D.new()
+## The default collider. It is automatically created and updated to match the size of the sprite
+var default_collider : CollisionShape2D
+## Whether or not the sprite is currently grabbed
+var is_grabbed := false :
+	set(value):
+		is_grabbed = value
+		if is_grabbed:
+			grabbed.emit()
+		else:
+			released.emit()
+
+
+# Helps a bit to make the dragging less choppy
+var grabbed_offset := Vector2.ZERO
+# Store for the original position
```

Comment:
This should be documented (and also moved to the 'variable declaration block' above), since I intend for `origin` to be accessible in case someone wants to change where the sprite returns to.

i.e. `origin` is not a purely internal variable

Reply:
In that case, making it an `export` would be the most logical thing to do!